django/forms/boundfield.py

```diff
@@ -260,6 +260,9 @@ def build_widget_attrs(self, attrs, widget=None):
                 attrs['required'] = True
         if self.field.disabled:
             attrs['disabled'] = True
+        id_for_label = self.id_for_label
+        if self.field.help_text and id_for_label and 'aria-describedby' not in widget.attrs:
+            attrs['aria-describedby'] = '%s_helptext' % id_for_label
```

Comment:
I find it problematic we’d make it possible to override the `aria-describedby` for two reasons:

- If the `help_text` is used, then I don’t think it would be appropriate for its content to be missing from the input’s description as computed from `aria-describedby`.
- Assuming we implement the other fix for #32819 by adding the field’s error(s) in `aria-describedby`, it would also be inappropriate for that to be missing because of a customization.

In both cases I guess this would be ok if the customization included the ids for the help text and error message when needed, but that doesn’t seem very convenient.

---

So we could make it possible to customize `aria-describedby`, but if we did in my opinion it should be in addition to Django automatically populating it for `help_text` and field errors:

```suggestion
        if self.field.help_text and id_for_label:
            helptext_id = '%s_helptext' % id_for_label
            if 'aria-describedby' in widget.attrs:
                attrs['aria-describedby'] = f"{helptext_id} {attrs['aria-describedby']}"
            else:
                attrs['aria-describedby'] = helptext_id
```

I’m not too sold on this either because then we’re having to assume the order of the different descriptions. It gets more problematic if we had errors in there too:

```python
attrs['aria-describedby'] = f"{helptext_id} {errors_id} {attrs['aria-describedby']}"
```

Reply:
Hi @thibaudcolas  -- Starting to think again. (Answers to where the last 6 months have gone on the back of a postcard).

Currently this PR adds `describedby` for help text. As you say likely we'll add a reference to errors too at some point. Do you see any other uses for `describedby`? 

I'm currently wondering if there is not, then we could add both in the same release and add a small breaking change note. 

If there's still the potential need to add additional references then I think we have to allow the customisation and add all of the options as you suggest.

I'd appreciate your thoughts.

Reply:
On reading the [MDN page](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby) there is another small issue using help text with `describedby`. It's [explicitly documented](https://docs.djangoproject.com/en/4.0/ref/models/fields/#help-text) that HTML is allowed in `help_text`, but it seems that the element pointed to by `describedby` [should contain only text](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby#description). As I understand it (I haven't tested it at all) assistive technologies are just supposed to use the string directly. We could instead use [`aria-details`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-details), although from reading on this it appears this is intended for more long-form information.

For your actual question, it is indeed intended for use for other things:

> The aria-describedby attribute is not limited to form controls. It can also be used to associate static text with widgets, groups of elements, regions that have a heading, definitions, and more.